v-next/hardhat/src/internal/builtin-plugins/network-manager/type-extensions/config.ts

```diff
@@ -33,16 +33,67 @@ declare module "../../../../types/config.js" {
     httpHeaders?: Record<string, string>;
   }
 
+  export type IntervalMiningConfig = number | [number, number];
+
+  export type MempoolOrder = "fifo" | "priority";
+
+  export type HardforkHistoryConfig = Map<
+    /* hardforkName */ string,
+    /* blockNumber */ number
+  >;
+
+  export interface HardhatNetworkChainConfig {
+    hardforkHistory: HardforkHistoryConfig;
+  }
+
+  export type HardhatNetworkChainsConfig = Map<
+    /* chainId */ number,
+    HardhatNetworkChainConfig
+  >;
+
+  export interface GenesisAccount {
+    privateKey: string;
+    balance: string | number | bigint;
+  }
+
+  export interface ForkConfig {
+    jsonRpcUrl: string;
+    blockNumber?: bigint;
+    httpHeaders?: Record<string, string>;
+  }
+
   export interface EdrNetworkUserConfig {
     type: "edr";
     chainId: number;
     chainType?: ChainType;
     from?: string;
-    gas: GasUserConfig;
-    gasMultiplier: number;
-    gasPrice: GasUserConfig;
+    // TODO: confirm these should be made optional
+    gas?: GasUserConfig;
+    gasMultiplier?: number;
+    gasPrice?: GasUserConfig;
```

Comment:
I made these optional to match the `HttpNetworkUserConfig`, it seemed odd that you couldn't just take the defaults for these values.